webapp/components/user_settings/user_settings_security.jsx

```diff
@@ -73,6 +73,15 @@ class SecurityTab extends React.Component {
         };
     }
 
+    componentWillUpdate() {
+        if ((this.state.passwordError !== '' || this.state.serverError) && (this.state.currentPassword !== '' || this.state.newPassword !== '' || this.state.confirmPassword !== '')) {
```

Comment:
That's one confusing looking if statement. Can you elaborate on the purpose and what it's doing?